java-checks/src/main/java/org/sonar/java/checks/spring/SpringAntMatcherOrderCheck.java

```diff
@@ -0,0 +1,185 @@
+/*
+ * SonarQube Java
+ * Copyright (C) 2012-2018 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.java.checks.spring;
+
+import com.google.common.annotations.VisibleForTesting;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Objects;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+import javax.annotation.CheckForNull;
+import org.sonar.check.Rule;
+import org.sonar.java.checks.helpers.ConstantUtils;
+import org.sonar.java.checks.methods.AbstractMethodDetection;
+import org.sonar.java.matcher.MethodMatcher;
+import org.sonar.java.matcher.TypeCriteria;
+import org.sonar.plugins.java.api.JavaFileScannerContext;
+import org.sonar.plugins.java.api.tree.ExpressionTree;
+import org.sonar.plugins.java.api.tree.MemberSelectExpressionTree;
+import org.sonar.plugins.java.api.tree.MethodInvocationTree;
+import org.sonar.plugins.java.api.tree.Tree;
+
+@Rule(key = "S4601")
+public class SpringAntMatcherOrderCheck extends AbstractMethodDetection {
+
+  private static final Pattern MATCHER_SPECIAL_CHAR = Pattern.compile("[?*{]");
+
+  private static final MethodMatcher AUTHORIZE_REQUESTS_MATCHER = MethodMatcher.create()
+    .typeDefinition(TypeCriteria.anyType())
+    .name("authorizeRequests")
+    .withAnyParameters();
+
+  private static final List<MethodMatcher> METHOD_MATCHERS = Collections.singletonList(
+    MethodMatcher.create()
+      .typeDefinition(TypeCriteria.subtypeOf("org.springframework.security.config.annotation.web.AbstractRequestMatcherRegistry"))
+      .name("antMatchers")
+      .withAnyParameters());
+
+  @Override
+  protected List<MethodMatcher> getMethodInvocationMatchers() {
+    return METHOD_MATCHERS;
+  }
+
+  @Override
+  protected void onMethodInvocationFound(MethodInvocationTree method) {
+    List<StringConstant> patterns = antMatchersPatterns(method);
+    if (!patterns.isEmpty()) {
+      List<StringConstant> previousPatterns = new PreviousMethodDetection().scan(method).previousPatterns;
```

Comment:
I would make the `scan` method static, in order to not have to create a new instance of `PreviousMethodDetection`. This can be hidden in the method. You could also directly return the `previousPatterns` rather than the instance.